src/mono/mono/component/mini-wasm-debugger.c

```diff
@@ -198,7 +199,14 @@ assembly_loaded (MonoProfiler *prof, MonoAssembly *assembly)
 		return;
 	}
 
-	if (mono_wasm_assembly_already_added(assembly->aname.name))
+	gboolean already_loaded = mono_bundled_resources_get_assembly_resource_values (assembly->aname.name, NULL, NULL);
```

Comment:
I think this previously also worked with assemblies downloaded by JS during application startup. I'm not sure this is still the case ?

Reply:
The `mono_wasm_assembly_already_added` would check for matches against the linked list `assemblies` [populated through `mono_wasm_add_assembly`](https://github.com/dotnet/runtime/blob/7b25983451968a7edfeba78a3a16c050dba12a4f/src/mono/wasm/runtime/driver.c#L194-L213). The new logic honors the prior code paths because `mono_wasm_add_assembly` will add the assembly information to the bundling hash table, and searches that hash table via apis like `mono_bundled_resources_get_assembly_resource_values`.
Is the intent of `mono_wasm_add_assembly` to not solely be used for bundling? I originally thought it was since the linked list `assemblies` is later [used to call `mono_register_bundled_assemblies`](https://github.com/dotnet/runtime/blob/7b25983451968a7edfeba78a3a16c050dba12a4f/src/mono/wasm/runtime/driver.c#L573C1-L583) in `mono_wasm_load_runtime`.

Would it be problematic to treat all instances of `mono_wasm_add_assembly` as bundling? Are there instances when `mono_wasm_add_assembly` is invoked after `mono_wasm_load_runtime` is done with the intent that those assemblies information would not be findable via bundling apis but findable through `mono_wasm_assembly_already_added`?

Reply:
> Is the intent of `mono_wasm_add_assembly` to not solely be used for bundling? 
> Would it be problematic to treat all instances of `mono_wasm_add_assembly` as bundling? 

I think you could unify/simplify it as you see fit. 
I didn't like scan of the linked list which was there, if I remember it right.